### DISCUSSION TITLE
Multi-step funnel visualization for goal conversions

### DISCUSSION THREAD

**Discussion by metmarkosaric** [plausible/analytics#171] [Anything Else]
"Display goal conversions with a multi-step funnel visualization to make it nicer visually but also to include more interesting data points such as percentage drop-off from step to step."

![mnhd4hvkenbghycqnczi](https://user-images.githubusercontent.com/34340819/85288482-12b27200-b496-11ea-8f28-aa0d3d3ac860.png)

**Update**: We've now opened up a preview version of funnels to everyone. You can now follow the visitor journey from a landing page to a conversion with funnel analysis. Details: https://plausible.io/docs/funnel-analysis

**Comment by metmarkosaric** [comment]
"On my blog, I have CTAs asking people to buy my ebook. When people click, they get sent to my sales page. What I would like to do is be able to track how many people click on the CTAs so I configured a goal. But I would also like to know how many people who click on the CTA end up buying the course. And I'm not sure how to set this up with Plausible. I have a "bought course" goal but it doesn't show me other goals that were completed before buying the course."

(feedback via Twitter DM)

**Reply by metmarkosaric** [reply]
One more vote via email today: "I am trying your software now and I was wondering how to set up a funnel? I want people to go from the landing page through several other pages, to the checkout and we could lose a lot of people at every step. Setting up a funnel is the first thing that anyone selling things on the web wants to do, so make it a priority."

**Reply by gustavgatu** [reply]
I couldn't agree more! Is this to expect anytime soon? (And does anyone know of any solutions meanwhile? I'm thinking maybe Klipfolio...?)

**Reply by metmarkosaric** [reply]
funnel visualization is something we plan to add but cannot make any promise on when. we are a small team with limited development resources so need to focus on one thing at a time. there are a few other more frequently requested features we have to complete first (cities/states, export/import...) before being able to start the work on this but you've been heard and it is something that is towards the top of our "features to build" list. stay tuned!

**Reply by ddahan** [reply]
@metmarkosaric Thanks for the explanation. In the meantime, can you describe an acceptable workaround? For example, if we define individual custom event goals, what's missing compared to a funnel?

**Reply by metmarkosaric** [reply]
custom events are acceptable workaround already. even when we introduce funnels, they will be based on custom events. you'll be allowed to choose which custom events are part of a funnel and which event is what step of a funnel. so all the data that we will show in funnels is there already as part of our current custom events and you can see it in the current dashboard. what you're missing is a nice visual chart that shows these different custom events in the correct order and with drop off percentages between them, you would need to manually calculate these now.

only new thing we might introduce with funnels is for them to be based on session tracking. for instance, if you're tracking a signup process with a natural order that all visitors go through, session based tracking won't make any difference compared to how custom events work now. but if you're tracking more random things that any visitor can start at any random step, then session based tracking would not consider those but will only consider those visitors that start with the correct step that you selected and that go through the funnel as you've set it up in the correct order. that will remove some of the noise but again not really important for signup processes and similar which is what most people use funnels for

**Reply by ddahan** [reply]
That's very clear, thank you very much 🙏 I'll start my signup analytics with custom events then, and will migrate as soon as available!

**Reply by metmarkosaric** [reply]
sounds like a plan, have fun!

**Reply by abrambailey** [reply]
I see states and exports are done now :)

**Comment by taylorlapeyre** [comment]
Just giving a huge +1 to this!

**Comment by modbp** [comment]
@metmarkosaric Hi there! Love Plausible so far. I'm on the last couple days of my trial and seriously considering becoming a paying user, but multi-step funnel visualization is the biggest thing missing for me so far.

I don't see this on the roadmap, but hoping you're planning to implement this soon. Any timeline or plans you can give an update on here? Thanks!

**Reply by metmarkosaric** [reply]
glad to hear that @modbp! funnel visualization is something we plan to add but cannot make any promise on when. we are a small team with limited development resources so need to focus on one thing at a time. there are a few other more frequently requested features we have to complete first (cities/states, export/import...) before being able to start the work on this but you've been heard and it is something that is towards the top of our "features to build" list. in terms of the timing of your trial expiring, it unfortunately will not be completed in the next couple of days but hope you do stay tuned!

**Reply by modbp** [reply]
Oh, ha, I wasn't expecting anything before the trial ended! 😂  Totally understand how it goes with the scarcity of time and the limited dev resources.

Glad to hear it is at least on the radar and even towards the top of the features list.

Thank you so much for the informative update and quick response!

**Comment by cosminstefanxp** [comment]
A huge 👍 to this from my side as well! For me, it's the only major feature I'm missing from other Analytics platforms, as it's important to figure out where people are dropped.

**Reply by teebu** [reply]
Any updates on this?

**Reply by metmarkosaric** [reply]
no updates on this unfortunately. we just completed GA import which was the biggest development work yet. we're focusing on [these items now](https://github.com/plausible/analytics/pulls). funnels is towards the top of new big features to build list alongside ecommerce metrics but i cannot make promise on when we'll be able to complete them. the data is already there if you set up custom events, we'll just create a nice graph for you to visualize the stats you can already have access to in your dashboard

**Comment by SemaphoreOxalis** [comment]
👍 on this We have to switch to another analitycs for some of our clients because of the this. Plausible is very adapted for small project like landing page, but funnel system is a mandatory function for these type of projects.

**Comment by Aulig** [comment]
This would be neat for projects with a long sign-up flow too. I'd also love seeing the average time it takes a user to go from one funnel step to the next one if that's possible.

**Comment by LauGau** [comment]
+1 here too! I would love to be able to show that nicely to my clients :)

**Comment by simon-guider** [comment]
+1

**Comment by martinzco** [comment]
Love it! Any timeline?

**Comment by ragamuf** [comment]
Uku touched on this roadmap subject in Episode 105 of [ThinkingElixir.com](https://podcasts.apple.com/us/podcast/thinking-elixir-podcast/id1516100616?i=1000567972204). It is worth a listen.

I have the same need and decided to generate funnels dynamically on the fly using the Stats API to drive feature-gated results and migrations. You can check out the details here. Maybe it can be of help while you wait.

[Programmatic Data Funnel Creation with Plausible Analytics and Elixir](https://gocassava.com/blog/programmatic-data-funnels-creation-with-plausible-analytics)

**Reply by martinzco** [reply]
Thanks for sharing. It's definitely an alternative yet not ideal.

**Comment by shylands** [comment]
+1, basically the only thing stopping me from fully switching over from GA

**Comment by abrambailey** [comment]
I guess one option is self hosting and querying your data using your data viz of choice ... but yeah, this is a huge one.

**Comment by metmarkosaric** [comment]
**Update**: We've now opened up a preview version of funnels to everyone. You can now follow the visitor journey from a landing page to a conversion with funnel analysis. Details: https://plausible.io/docs/funnel-analysis

**Reply by melakhvision** [reply]
@metmarkosaric is this feature available on the free version.
I have a self hosted with the latest docker image but I don't see that feature. Can clarify ?


![funnel missing](https://github.com/plausible/analytics/assets/45154979/fb9bef95-3f49-46a5-898b-fd31672945f0)

**Reply by Igloczek** [reply]
> Funnel analysis is an upcoming premium feature that's free-to-use during the private preview. Pricing will be announced soon.

So this project will not be 100% open-source soon, or by the "premium" you mean the cloud?

**Reply by samipe** [reply]
The funnel feature seems to work nicely, but it's missing a few functions that I'd need to use it in my projects:

1. Chaining goals with AND only is not enough. Ideally there would be support for something like `visit /shop AND (navbar click OR featured product click)`

2. I've implemented custom properties in quite a lot of events. Sadly these properties aren't integrated with funnel setup **or** goal setup. I'd like to either...
    A. Setup funnels with filtered goals such as `navbar click {element=dropdown}`
    B. Setup more advanced custom goals with selectors like `navbar click {element=cart OR element=profile}`

**Comment by zemlanin** [comment]
Is it possible to analyze multiple entry points to the funnel? For example, "from what page did visitor arrive on `/register`"

"Filter by a goal and check Entry pages" is close, but my understanding is that Entry pages are for the session's first page view, so the `/ -> /pricing -> /register` would look the same as `/ -> /register`?..

_Sorry for digging up an old discussion_
